File: 0001-two-sum/solution.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution
{
public:
    vector<int> twoSum(vector<int> &nums, int target)
    {
        int arr[2];
        for (int i = 0; i < (int)nums.size(); i++)
        {
            for (int j = i+1; j < (int)nums.size(); j++)
            {
                if (nums.at(i) + nums.at(j) == target)
                {

                    arr[0] = i;
                    arr[1] = j;
                    goto out;
                }
            }
        }
    out:
        vector<int> v(arr, arr + 2);
        return v;
    }
};

File: 0001-two-sum/solution.py
class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        # return index
        mapy = {}

        for i, num in enumerate(nums):
            looking = target - num

            if looking in mapy:
                return [mapy[looking], i]

            mapy[num] = i

        return None

